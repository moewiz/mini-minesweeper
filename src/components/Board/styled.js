import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

export const BoardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-top: 30px;
`;

export const LoadingStyled = styled.span`
  font-size: 20px;
`;

export const ErrorStyled = styled.span`
  font-size: 20px;
  color: #ff4141;
`;

export const HeaderWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 30px;
`;

const HeaderBtn = css`
  color: #000;
  cursor: pointer;
  font-size: 20px;
  padding: 10px;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`;

export const NewGameStyled = styled.div`
  ${HeaderBtn}
`;

export const HomeStyled = styled(Link)`
  ${HeaderBtn}
`;
