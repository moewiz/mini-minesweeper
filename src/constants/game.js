export const BEGINNER = {
  size: 9,
  mines: 10
};

export const ADVANTAGE = {
  size: 16,
  mines: 40
};

export const GAME_STATUS = {
  LOSE: -1,
  NOT_START: 0,
  PLAYING: 1,
  WIN: 2
};
