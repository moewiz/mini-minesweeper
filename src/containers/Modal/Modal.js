import React from "react";
import { useLocation } from "react-router-dom";
import dayjs from "dayjs";
import { Modal } from "../../components";
import {
  actions as MiniMineSweeperActions,
  selectors as MiniMineSweeperSelectors
} from "../../reducers/miniMineSweeper";
import { useActions, useShallowEqualSelector } from "../../utils/hookRecipes";

const addZero = n => (n < 10 ? `0${n}` : n);

const ModalContainer = () => {
  const [setupBeginner, setupAdvanced] = useActions([
    MiniMineSweeperActions.setupBeginnerGame,
    MiniMineSweeperActions.setupAdvantageGame
  ]);
  const { pathname } = useLocation();
  let setupGame = () => null;
  const gameStatus = useShallowEqualSelector(
    MiniMineSweeperSelectors.getGameStatus
  );
  const startTime = useShallowEqualSelector(
    MiniMineSweeperSelectors.getStartTime
  );
  const seconds = startTime ? dayjs().diff(startTime, "second") : 0;
  const hh = addZero(parseInt(seconds / 3600, 10));
  const mm = addZero(parseInt((seconds / 60) % 60, 10));
  const ss = addZero(parseInt(seconds % 60, 10));
  const duration = `${hh}:${mm}:${ss}`;

  if (pathname.includes("beginner")) setupGame = setupBeginner;
  else setupGame = setupAdvanced;

  return (
    <Modal gameStatus={gameStatus} setupGame={setupGame} duration={duration} />
  );
};

export default ModalContainer;
