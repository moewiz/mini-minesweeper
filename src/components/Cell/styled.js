import styled, { css } from "styled-components";

export const CellWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  border: 1px solid;
  border-color: #000;
  background: ${props => (props.isOpen ? "#e4e4e4" : "#fff")};
  font-weight: 600;
  ${props =>
    props.isMine &&
    props.isOpen &&
    css`
      position: relative;
      &:after {
        content: "";
        position: absolute;
        background: #24292e !important;
        border: 1px solid #24292e;
        border-radius: 50%;
        top: 4px;
        left: 4px;
        width: 20px;
        height: 20px;
      }
    `}
  &:hover {
    background: #e4e4e4;
  }
`;
