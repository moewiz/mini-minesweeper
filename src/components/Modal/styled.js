import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

export const ButtonWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const HeaderBtn = css`
  color: #000;
  cursor: pointer;
  font-size: 20px;
  padding: 10px;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`;

export const NewGameStyled = styled.div`
  ${HeaderBtn}
`;

export const HomeLinkStyled = styled(Link)`
  ${HeaderBtn}
`;

export const MessageStyled = styled.div`
  font-size: 20px;
  text-align: center;
  margin-top: 5px;
`;
