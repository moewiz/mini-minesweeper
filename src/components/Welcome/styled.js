import styled from "styled-components";
import { Link } from "react-router-dom";

export const HomeWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-top: 30px;
`;

export const GroupLevelStyled = styled.div`
  display: flex;
  flex-direction: row;
  width: 350px;
  justify-content: space-around;
`;

export const LevelBtnStyled = styled(Link)`
  font-size: 20px;
  color: #000;
  border: 1px solid;
  padding: 15px;
  text-decoration: none;
  display: flex;
  &:hover {
    text-decoration: underline;
  }
`;
