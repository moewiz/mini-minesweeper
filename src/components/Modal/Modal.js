import React from "react";
import PropTypes from "prop-types";
import Popup from "reactjs-popup";
import { GAME_STATUS } from "../../constants/game";
import {
  ButtonWrapper,
  NewGameStyled,
  HomeLinkStyled,
  MessageStyled
} from "./styled";

const Modal = ({ gameStatus, duration, setupGame }) => {
  return (
    <Popup
      open={gameStatus === GAME_STATUS.WIN || gameStatus === GAME_STATUS.LOSE}
    >
      <MessageStyled>
        {gameStatus === GAME_STATUS.LOSE ? "💀" : "🏆"}
      </MessageStyled>
      <MessageStyled>
        You {gameStatus === GAME_STATUS.LOSE ? "lost" : "won"} the game in{" "}
        {duration}
      </MessageStyled>
      <ButtonWrapper>
        <NewGameStyled onClick={setupGame}>New game</NewGameStyled>
        <HomeLinkStyled to="/">Home page</HomeLinkStyled>
      </ButtonWrapper>
    </Popup>
  );
};

Modal.propTypes = {
  gameStatus: PropTypes.number.isRequired,
  duration: PropTypes.string.isRequired,
  setupGame: PropTypes.func.isRequired
};

export default Modal;
