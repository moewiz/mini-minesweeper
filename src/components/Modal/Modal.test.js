import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router, Link } from "react-router-dom";
import Modal from "./Modal";
import { NewGameStyled } from "./styled";
import { GAME_STATUS } from "../../constants/game";

describe("<Modal />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const setupGameMock = jest.fn();

  const Component = mount(
    <Router>
      <Modal
        duration="00:00:00"
        gameStatus={GAME_STATUS.WIN}
        setupGame={setupGameMock}
      />
    </Router>
  );

  it("should render without thrown an error", () => {
    expect(Component.length).toBe(1);
  });

  it("match snapshot", () => {
    expect(Component).toMatchSnapshot();
  });

  it("should trigger setupGameMock when New button is clicked", () => {
    const NewGameBtn = Component.find(NewGameStyled).at(0);
    NewGameBtn.simulate("click");
    expect(setupGameMock).toBeCalledTimes(1);
  });

  it("should render 1 Link", () => {
    const Links = Component.find(Link);
    expect(Links.length).toBe(1);
  });
});
