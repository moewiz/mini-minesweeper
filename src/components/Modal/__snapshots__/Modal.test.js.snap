// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Modal
      duration="00:00:00"
      gameStatus={2}
      setupGame={[MockFunction]}
    >
      <Popup
        arrow={true}
        arrowStyle={Object {}}
        className=""
        closeOnDocumentClick={true}
        closeOnEscape={true}
        contentStyle={Object {}}
        defaultOpen={false}
        disabled={false}
        keepTooltipInside={false}
        lockScroll={false}
        modal={false}
        mouseEnterDelay={100}
        mouseLeaveDelay={100}
        offsetX={0}
        offsetY={0}
        on={
          Array [
            "click",
          ]
        }
        onClose={[Function]}
        onOpen={[Function]}
        open={true}
        overlayStyle={Object {}}
        position="bottom center"
        repositionOnResize={true}
        trigger={null}
      >
        <div
          key="H"
          style={
            Object {
              "left": "0px",
              "position": "absolute",
              "top": "0px",
            }
          }
        />
        <div
          className="popup-overlay "
          key="O"
          onClick={[Function]}
          style={
            Object {
              "background": "rgba(0, 0, 0,0.5)",
              "bottom": "0",
              "display": "flex",
              "left": "0",
              "position": "fixed",
              "right": "0",
              "top": "0",
              "zIndex": "999",
            }
          }
        >
          <div
            className="popup-content "
            key="C"
            onClick={[Function]}
            style={
              Object {
                "background": "rgb(255, 255, 255)",
                "border": "1px solid rgb(187, 187, 187)",
                "margin": "auto",
                "padding": "5px",
                "position": "relative",
                "width": "50%",
              }
            }
          >
            <div>
              <styled.div>
                <div
                  className="sc-AxhCb kCEjYJ"
                >
                  üèÜ
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-AxhCb kCEjYJ"
                >
                  You 
                  won
                   the game in
                   
                  00:00:00
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-AxjAm bGbvop"
                >
                  <styled.div
                    onClick={[MockFunction]}
                  >
                    <div
                      className="sc-AxirZ hIyvNJ"
                      onClick={[MockFunction]}
                    >
                      New game
                    </div>
                  </styled.div>
                  <Styled(Link)
                    to="/"
                  >
                    <Link
                      className="sc-AxiKw dVolwd"
                      to="/"
                    >
                      <LinkAnchor
                        className="sc-AxiKw dVolwd"
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          className="sc-AxiKw dVolwd"
                          href="/"
                          onClick={[Function]}
                        >
                          Home page
                        </a>
                      </LinkAnchor>
                    </Link>
                  </Styled(Link)>
                </div>
              </styled.div>
            </div>
          </div>
        </div>
      </Popup>
    </Modal>
  </Router>
</BrowserRouter>
`;
